src/languages/fsharp.js

```diff
@@ -94,30 +99,21 @@ export default function(hljs) {
     illegal: /\/\*/,
     contains: [
       {
-        // monad builder keywords (matches before non-bang kws)
-        className: 'keyword',
-        begin: /\b(yield|return|let|do)!/
-      },
-      {
-        className: 'string',
-        begin: '@"',
-        end: '"',
-        contains: [
-          {
-            begin: '""'
-          }
-        ]
+        // monad builder keywords (matches before non-bang keywords)
+        scope: 'keyword',
+        match: /\b(yield|return|let|do|match|use)!/
       },
       {
-        className: 'string',
-        begin: '"""',
-        end: '"""'
+        scope: 'string',
+        // matches triple quote strings, verbatim strings (@""), character literals...
+        match: /(?:"""[\s\S]*?"""|@"(?:""|[^"])*"|"(?:\\[\s\S]|[^\\"])*")|'(?:[^\\']|\\(?:.|\d{3}|x[a-fA-F\d]{2}|u[a-fA-F\d]{4}|U[a-fA-F\d]{8}))'/
```

Comment:
We really need to use some of our regex tools (or `variants`) to make this easier to read... we try not to encourage large unreadable regex in the codebase - we'd rather have the rules written out separately for 10 types of strings so it's easier to maintain, etc.

Reply:
Things like escaping inside strings would likely be matched in a `contains` mode, not inside the primary regex.